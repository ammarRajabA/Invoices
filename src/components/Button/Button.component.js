import React, { Component } from 'react';

import './Button.style.css'

export default class Button extends Component{
	render(){
		return(
			<button className={`button ${this.props.type}`} {...this.props} onClick={this.props.type==="disabled"?()=>{}:this.props.onClick}>
				{this.props.icon&&<i className={this.props.icon}></i>}
				{this.props.label&&<span>{this.props.label}</span>}
			</button>
			)
	}
}