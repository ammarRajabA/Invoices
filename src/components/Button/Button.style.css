.button{
	border-radius: 5px;
	background-color: #4D87EC;
	padding-left: 15px;
	padding-right: 15px;
	padding-top: 10px;
	padding-bottom: 10px;
	box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2), 0 3px 10px 0 rgba(0,0,0,0.19);
	cursor: pointer;
	font-size: 18px;
	border-width:1px;
	border-color: #3C76DB;
	border-style: solid;
	transition: 0.3s;
	margin-left: 20px;
	margin-right: 20px;
	font-family: calibri
}
.button:hover{
	background-color: #3C76DB;
}
.button span{
	margin-left: 10px;
	margin-right: 10px;
	color: white;
}
.button i{
	color: white;
}

.warning{
	background-color: #FFCC00;
	border-color: #EECC00;
}
.warning:hover{
	background-color: #EECC00;
}

.danger{
	background-color: #D9534F;
	border-color: #C8423E;
}
.danger:hover{
	background-color: #C8423E;
}

.success{
	background-color: #61B2A7;
	border-color: #50A196;
}
.success:hover{
	background-color: #50A196;
}

.disabled{
	background-color: #B5B5B5;
	border-color: #A4A4A4;
}
.disabled:hover{
	background-color: #A4A4A4;
}
.solid{
	background-color: #FFFFFF;
	border-color: #FFFFFF;
}
.solid:hover{
	background-color: #FFFFFF;
}
.solid span{
	color:#D5D5D5;
}
.solid i{
	color: #D5D5D5;
}

.normal{
	background-color: #FFFFFF;
	border-color: #FFFFFF;
}
.normal:hover{
	background-color: #FFFFFF;
}
.normal span{
	color:#131313;
}
.normal i{
	color: #131313;
}