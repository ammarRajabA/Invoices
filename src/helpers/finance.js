export const calculateTotalPriceWithTax=(invoiceData)=>{
	var total=0.0;
	if (Array.isArray(invoiceData)){
		invoiceData.map((invoice)=>{
			total+=calculateTotalPriceWithTax(invoice)
		})
		return total
	}else{
		invoiceData.items.map((item)=>{
			total+=item.qty*item.unitPriceNet*(1+parseFloat(item.taxRate))
		})
		return total
	}
}
export const calculateTotalPriceNoTax=(invoiceData)=>{
	var total=0.0;
	if (Array.isArray(invoiceData)){
		invoiceData.map((invoice)=>{
			total+=calculateTotalPriceWithTax(invoice)
		})
		return total
	}else{
		invoiceData.items.map((item)=>{
			total+=item.qty*item.unitPriceNet
		})
		return total
	}
}

export const calculateTotalItemNoTax=(item)=>item.qty*item.unitPriceNet
export const calculateTotalItemWithTax=(item)=>item.qty*item.unitPriceNet*(1+parseFloat(item.taxRate))