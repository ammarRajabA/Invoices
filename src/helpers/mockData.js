import faker from 'faker';

function randomInvoice() {
  const senderName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  const recipientName = `${faker.name.firstName()} ${faker.name.lastName()}`;
  return {
    "id": faker.random.uuid(),
    "dueDate": faker.date.future(),
    "createdDate": faker.date.past(),
    "sender": {
      "name": senderName,
      "email": `${senderName.split(" ").map(n => n.toLowerCase()).join(".")}@sender.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "recipient": {
      "name": recipientName,
      "email": `${recipientName.split(" ").map(n => n.toLowerCase()).join(".")}@recipient.com`,
      "address1": `${faker.company.companyName()} ${faker.company.companySuffix()}`,
      "address2": `${faker.address.streetName()}, ${faker.random.number({min: 1, max: 100})}`,
      "address3": `${faker.address.zipCode()} ${faker.address.city()}`,
      "vatId": faker.helpers.replaceSymbolWithNumber("DE#########")
    },
    "items": Array.apply(null, new Array(faker.random.number({min: 1, max: 20}))).map(() => (
      {
        "description": faker.commerce.productName(),
        "qty": faker.random.number({min: 1, max: 10}),
        "unitPriceNet": parseFloat(faker.finance.amount(0.01,100,2)),
        "taxRate": 0.19
      }
    ))
  }
}

const NUM_INVOICES = 10;
const INVOICES = Array.apply(null, new Array(NUM_INVOICES)).map(randomInvoice);

export default {
	invoices:INVOICES,
	meta:{
		total:100,
		totalPages:10,
		pageNumber:1,
		pageSize:10
	},
  INIT_STATE:{
    invoices:[],
    selectedInvoice:null,
    loading:false,
    invoiceSelected:null,
    meta:{
      total:0,
      totalPages:0,
      pageNumber:0,
      pageSize:0
    },
    newInvoice:{
      "id": -1,
        "dueDate": new Date(),
        "createdDate": new Date(),
        "sender": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "recipient": {
            "name": '',
            "email": '',
            "address1": '',
            "address2": '',
            "address3": '',
            "vatId": ''
        },
        "items": [{description:'',qty:0,unitPriceNet:0,taxRate:0}]
    }
  }
}