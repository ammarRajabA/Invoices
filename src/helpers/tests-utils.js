import React from 'react'
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

export const makeMountRender = (Component, defaultProps = {}) => {
	return (customProps={})=>{
		const props={...defaultProps,...customProps};
		return mount(<Component {...props}/>)
	}
}

export const reduxify = (Component, props = {}, store= {}) => {
	return function reduxWrap() {
		return (
			<Provider store={store}>
				<Component {...props} />
			</Provider>
		)
	}
}

export const snapshotify = reactWrapper => {
	return reactWrapper.debug();
};