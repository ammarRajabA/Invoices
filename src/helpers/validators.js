export const validateEmail=(email)=>{
	var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}
export const validateVATID=(vatID)=>{
	return /(DE)?[0-9]{9}/.test(vatID)
}