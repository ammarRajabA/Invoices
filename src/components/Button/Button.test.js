import { makeMountRender, snapshotify } from '../../helpers/tests-utils';

import Button from './Button.component'

describe('Button component',function () {
	it('match snapshot',function () {
		const wrapper = makeMountRender(Button)();
		expect(snapshotify(wrapper)).toMatchSnapshot();
	})
	it('should execute function onClick',function () {
		const wrapper = makeMountRender(Button)({onClick:()=>"This is result"});
		expect(wrapper.props().onClick()).toBe("This is result")
	})
	it('should has class danger when passing danger type',function () {
		const wrapper = makeMountRender(Button)({type:"danger"});
		expect(wrapper.find('button').hasClass('button danger')).toBe(true)
	})
})