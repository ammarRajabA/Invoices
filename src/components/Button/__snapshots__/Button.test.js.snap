// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component match snapshot 1`] = `
"<Button>
  <button className=\\"button undefined\\" onClick={[undefined]} />
</Button>"
`;
